src/lib/logger.js

```diff
@@ -20,61 +20,27 @@
 */
 
 import { Container } from 'constitute';
-
-import chalk from 'chalk';
-import settings from '../settings';
-
-function isObject(obj: any): boolean {
-  return obj === Object(obj);
-}
-
-function _transform(...params: Array<any>): Array<any> {
-  return params.map((param: any): string => {
-    if (!isObject(param)) {
-      return param;
-    }
-
-    return JSON.stringify(param);
-  });
-}
-
-function getDate(): string {
-  return (new Date()).toISOString();
-}
-
-class Logger {
-  static container: Container;
-
-  static log(...params: Array<any>) {
-    if (settings.SHOW_VERBOSE_DEVICE_LOGS) {
-      Logger._log(`[${getDate()}]`, _transform(...params));
-    }
-  }
-
-  static info(...params: Array<any>) {
-    Logger._log(
-      `[${getDate()}]`,
-      chalk.cyan(_transform(...params)),
-    );
-  }
-
-  static warn(...params: Array<any>) {
-    Logger._log(
-      `[${getDate()}]`,
-      chalk.yellow(_transform(...params)),
-    );
-  }
-
-  static error(...params: Array<any>) {
-    Logger._log(
-      `[${getDate()}]`,
-      chalk.red(_transform(...params)),
-    );
-  }
-
-  static _log(...params: Array<any>): Function {
-    return Logger.container.constitute('LOGGING_FUNCTION')(...params);
-  }
-}
-
-export default Logger;
+import { Logger as DefaultLogger } from './defaultLogger';
+import { ILogger } from '../types';
+
+
+let logger : ILogger = DefaultLogger;
```

Comment:
Nope, it will throw warning if you change this to const, cause the logger is assigned later, if the container is used to initialize. This instance here is only called, if a logger is used before/during container initialization (which can happen, cause the logger is also used in process.onUnCaughtexceptuon)
 But i will change the Implementation as jlkalberer suggested, so maybe this will be changed